runtime/src/accounts.rs

```diff
@@ -581,6 +569,39 @@ impl Accounts {
     }
 }
 
+fn collect_accounts<'a>(
```

Comment:
You certain this explicit lifetime is required? and that the same lifetime would be required for many of the parameters? Looks over-constrained.

Reply:
I sure haven't found a way to compile without the explicit lifetime. Happy to look at a different pattern, if you have ideas.